client/src/modules/dashboard/Events/pages/EventsPage.tsx

```diff
@@ -21,70 +22,120 @@ import { useUser } from '../../../auth/user';
 import { useDashboardEventsQuery } from '../../../../generated/graphql';
 import { NextPageWithLayout } from '../../../../pages/_app';
 
-const ShowCanceledSwitch = ({
-  setShowCanceled,
-  defaultChecked,
-}: {
-  setShowCanceled: React.Dispatch<React.SetStateAction<boolean>>;
+interface FilterEventsProps {
+  setFilterEvent: React.Dispatch<React.SetStateAction<boolean>>;
   defaultChecked: boolean;
-}) => {
+  filterLabel: string;
+  id: string;
+}
+
+const FilterEvents = ({
+  setFilterEvent,
+  defaultChecked,
+  filterLabel,
+  id,
+}: FilterEventsProps) => {
   return (
-    <Flex>
-      <FormLabel htmlFor="show-canceled-events">Show canceled events</FormLabel>
+    <>
+      <FormLabel marginTop=".5em" htmlFor={id}>
+        {filterLabel}
+      </FormLabel>
       <Switch
         isChecked={defaultChecked}
-        id="show-canceled-events"
-        onChange={(e) => setShowCanceled(e.target.checked)}
+        id={id}
+        onChange={(e) => setFilterEvent(e.target.checked)}
       />
-    </Flex>
+    </>
```

Comment:
The `Switch` component is weird, it has `label` built in it, but it doesn't have context inside of it, so currently we have two `label` elements.


We can change it to this, but I don't think this is accessible, because the context come after the `input`. here is the sourse code 

https://github.com/chakra-ui/chakra-ui/blob/3e1b3f6b6a7398b71ac08339110f075695fbae94/packages/components/switch/src/switch.tsx#L82-L109

```ts
      <Switch
        marginTop='.25rem'
        htmlFor={id}
        isChecked={defaultChecked}
        id={id}
        onChange={(e) => setFilterEvent(e.target.checked)}
        fontWeight='500'
      >
        {filterLabel}
      </Switch>
```

Reply:
![image](https://user-images.githubusercontent.com/88248797/216596964-4ae9f512-f424-43e3-8620-53487af1217f.png)